qos-monitor/src/main/java/eu/arrowhead/core/qos/service/ping/monitor/impl/DefaultExternalPingMonitor.java

```diff
@@ -0,0 +1,204 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.util.Assert;
+import org.springframework.web.util.UriComponents;
+
+import eu.arrowhead.common.CommonConstants;
+import eu.arrowhead.common.CoreCommonConstants;
+import eu.arrowhead.common.Utilities;
+import eu.arrowhead.common.dto.shared.IcmpPingRequestACK;
+import eu.arrowhead.common.dto.shared.IcmpPingRequestDTO;
+import eu.arrowhead.common.dto.shared.SystemRequestDTO;
+import eu.arrowhead.common.exception.ArrowheadException;
+import eu.arrowhead.core.qos.QosMonitorConstants;
+import eu.arrowhead.core.qos.dto.IcmpPingResponse;
+import eu.arrowhead.core.qos.service.QoSMonitorDriver;
+import eu.arrowhead.core.qos.service.ping.monitor.AbstractPingMonitor;
+import eu.arrowhead.core.qos.service.ping.monitor.PingEventCollectorTask;
+import eu.arrowhead.core.qos.service.ping.monitor.PingEventProcessor;
+
+public class DefaultExternalPingMonitor extends AbstractPingMonitor{
+
+	//=================================================================================================
+	// members
+
+	//-------------------------------------------------------------------------------------------------
+	private static final int ICMP_TTL = 255;
+	private static final int OVERHEAD_MULTIPLIER = 2;
+
+	private final ThreadPoolExecutor threadPool = (ThreadPoolExecutor) Executors.newFixedThreadPool(1);
+
+	@Autowired
+	private QoSMonitorDriver driver;
+
+	@Autowired
+	private PingEventProcessor processor;
+
+	@Resource(name = QosMonitorConstants.EVENT_COLLECTOR)
+	private PingEventCollectorTask eventCollector; 
+
+	@Value(CoreCommonConstants.$QOS_MONITOR_PROVIDER_NAME_WD)
+	private String externalPingMonitorName;
+
+	@Value(CoreCommonConstants.$QOS_MONITOR_PROVIDER_ADDRESS_WD)
+	private String externalPingMonitorAddress;
+
+	@Value(CoreCommonConstants.$QOS_MONITOR_PROVIDER_PORT_WD)
+	private int externalPingMonitorPort;
+
+	@Value(CoreCommonConstants.$QOS_MONITOR_PROVIDER_PATH_WD)
+	private String externalPingMonitorPath;
+
+	@Value(CoreCommonConstants.$QOS_MONITOR_PROVIDER_SECURE_WD)
+	private boolean pingMonitorSecure;
+
+	@Value(CoreCommonConstants.$QOS_MONITOR_PROVIDER_AUTHINFO_WD)
+	private String externalPingMonitorAuthInfo;
+
+	private SystemRequestDTO pingMonitorSystem;
+
+	private final Logger logger = LogManager.getLogger(DefaultExternalPingMonitor.class);
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	@Override
+	public List<IcmpPingResponse> ping(final String address) {
+		logger.debug("ping statred...");
+
+		final int timeOut = calculateTimeOut();
+		final UUID measurementProcessId = requestExternalMeasurement(address);
+
+		final long startTime = System.currentTimeMillis();
+		final long meausermentExpiryTime = startTime + timeOut;
```

Comment:
typo in variable name: meausermentExpiryTime  => measurementExpiryTime

Reply:
fixed